src/components/reusable/datePicker/datepicker.ts

```diff
@@ -79,135 +167,327 @@ export class DatePicker extends FormMixin(LitElement) {
+        </div>
 
-        <div
-          class="${classMap({
-            'input-wrapper': true,
-          })}"
-        >
+        <div class="input-wrapper">
           <input
-            class="${classMap({
-              'size--sm': this.size === 'sm',
-              'size--lg': this.size === 'lg',
-            })}"
-            datePickerType=${this.datePickerType}
-            type=${this.datePickerType === DATE_PICKER_TYPES.WITHITIME
-              ? 'datetime-local'
-              : 'date'}
-            id=${this.name ? this.name : 'datepicker-1'}
-            name=${this.name}
-            value=${this.value}
+            class="input-custom"
+            type="text"
+            id=${anchorId}
+            name=${this.nameAttr}
+            placeholder=${placeholder}
+            ?disabled=${this.datePickerDisabled}
             ?required=${this.required}
-            ?disabled=${this.disabled}
             ?invalid=${this._isInvalid}
-            aria-invalid=${this._isInvalid}
-            aria-describedby=${this._isInvalid
-              ? 'error'
-              : this.warnText !== '' && !this._isInvalid
-              ? 'warning'
-              : ''}
-            min=${ifDefined(this.minDate)}
-            max=${ifDefined(this.maxDate)}
-            step=${ifDefined(this.step)}
-            @input=${(e: any) => this.handleInput(e)}
+            aria-invalid=${this._isInvalid ? 'true' : 'false'}
+            aria-labelledby=${`label-${anchorId}`}
+            @click=${this.handleInputClickEvent}
+            @focus=${this.handleInputFocusEvent}
           />
+          <span class="icon">${unsafeSVG(calendarIcon)}</span>
         </div>
-        ${this.caption !== ''
-          ? html` <div class="caption">${this.caption}</div> `
-          : null}
-        ${this._isInvalid
-          ? html`
-              <div id="error" class="error">
-                ${this.invalidText || this._internalValidationMsg}
-              </div>
-            `
-          : null}
-        ${this.warnText !== '' && !this._isInvalid
-          ? html`<div id="warning" class="warn">${this.warnText}</div>`
-          : null}
+
+        ${this.caption
+          ? html`<div
+              id=${descriptionId}
+              class="caption"
+              @mousedown=${this.preventFlatpickrOpen}
+              @click=${this.preventFlatpickrOpen}
+            >
+              ${this.caption}
+            </div>`
+          : ''}
+        ${this.renderValidationMessage(errorId, warningId)}
       </div>
     `;
   }
 
-  // calls when start date or value change
-  private handleInput(e: any) {
-    this.value = e.target.value;
+  private renderValidationMessage(errorId: string, warningId: string) {
+    if (this.invalidText || (this._isInvalid && this._hasInteracted)) {
+      return html`<div
+        id=${errorId}
+        class="error error-text"
+        role="alert"
+        aria-label=${this.errorAriaLabel || 'Error message'}
```

Comment:
This aria-label is probably only needed on the icon, since the div also has visible text.

Reply:
✅